pkg/loki/client_test.go

```diff
@@ -0,0 +1,48 @@
+package loki_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/flared/lokify/pkg/loki"
+)
+
+func TestQuery(t *testing.T) {
+	respBody := &loki.QueryResponse{
+		Status: "success",
+		Data: loki.QueryData{
+			ResultType: "streams",
+			Result: []loki.QueryDataResult{
+				{
+					Stream: map[string]string{"label": "value"},
+					Values: [][2]string{{"123456789", "log message"}},
+				},
+			},
+		},
+	}
+	respBodyBytes := new(bytes.Buffer)
```

Comment:
```
var respBodyBytes bytes.Buffer
```

Reply:
json.NewEncoder(buffer) doesn't seem to accept nil

```
	var respBodyBytes bytes.Buffer
	if err := json.NewEncoder(respBodyBytes).Encode(respBody); err != nil {
		t.Fatal(err)
	}

output:
var respBodyBytes bytes.Buffer
cannot use respBodyBytes (variable of type bytes.Buffer) as io.Writer value 
in argument to json.NewEncoder: missing method Write 
(Write has pointer receiver) compilerInvalidIfaceAssign)
```

Reply:
hmm maybe just `b := &bytes.Buffer{}`

Reply:
or `var b *bytes.Buffer`

Reply:
`&bytes.Buffer{}` is better then `new(bytes.Buffer)` ?

Reply:
ok you are right !
this is working:
```
var b *bytes.Buffer
```

Thanks!

Reply:
oh never mind

```
--- FAIL: TestQuery (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4e70e6]
```